runner.js

```diff
@@ -0,0 +1,71 @@
+module.exports = class Runner {
+  constructor(it, steps, deepEqual) {
+    this.it = it;
+    this.steps = steps;
+    this.deepEqual = deepEqual;
+    this.results = [];
+    this.isDone = false;
+  }
+
+  run() {
+    const { it, steps } = this;
+
+    steps.reduce((prevResult, step) => {
+      if (this.isDone) throwExtraStep(step);
+
+      let output;
+      try {
+        output = runStep(it, step, prevResult);
+        runAssertions(step, output, this.deepEqual);
+      } catch (err) {
+        err.stack = `\n\nStack:\n${step.stack}`;
```

Comment:
Replacing the stack with the stack of the step call. (where `.yield(...)` gets called )